sematic/metrics/func_success_rate_metric.py

```diff
@@ -0,0 +1,77 @@
+# Standard Library
+from datetime import datetime
+from typing import Optional, Tuple
+
+# Third-party
+from sqlalchemy.orm import Query, Session, joinedload
+
+# Sematic
+from sematic.abstract_future import FutureState
+from sematic.abstract_system_metric import AbstractSystemMetric
+from sematic.db.models.run import Run
+from sematic.metrics.metric_point import MetricType
+
+FINAL_STATES = {FutureState.RESOLVED, FutureState.FAILED, FutureState.NESTED_FAILED}
+
+
+class FuncSuccessRateMetric(AbstractSystemMetric):
+    """
+    System Metric to measure Sematic Function's success rate.
+
+    Success rate is defined as:
+
+    Numerator: Number of qualifying RESOLVED runs
+    Denominator: Number of qualifying (RESOLVED + FAILED + NESTED_FAILED) runs
+
+    where qualifying runs are all non-cached, non-cloned runs.
```

Comment:
The "do we include cached/cloned?" question is actually an interesting one. If I execute a pipeline and it only executes `foo` 20 times, but uses a cached value for `foo` 80 times, then 90 times out of 100 I got the result I wanted. If my question is "how often can I run `foo` and not have to worry about fixing something?" then 90% is a good answer. If my question is "is the code for `foo` robust?" then 50% is a good answer.

Reply:
Perhaps we should ask users which they would want to see more